helm/cluster-cloud-director/templates/_helpers.tpl

```diff
@@ -219,3 +220,54 @@ taints:
 {{- $outerScope := . }}
 {{- include "mtRevision" (merge (dict "currentClass" .Values.controlPlane) $outerScope.Values) }}
 {{- end -}}
+
+{{/*
+Generate a stanza for KubeAdmConfig and KubeAdmControlPlane in order to 
+mount containerd configuration for registry configuration in nodes.
+*/}}
+{{- define "registryFiles" -}}
+{{- if .Values.registry.configure -}}
+- path: /etc/containerd/conf.d/registry-config.toml
```

Comment:
I have whitnesed an environment where the `/etc/containerd/conf.d/` directory was not created. `containerd` accepted the config if I created the dir and then placed the cfg file into that directory, but wasn't there from the beginning. I had to create the dir manually. Just a heads up, I am not sure if this `path: /etc/containerd/conf.d/registry-config.toml` construct also creates all the dirs on the way.. like `mkdir -p ..`

It might be related to `containerd` version.

Reply:
We haven't hit any issues regarding this until now. I assume it runs `mkdir -p`.